### DISCUSSION TITLE
Why is code throwing inside async CE uncaught by try statement

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/fsharp#14982] [Q&A]
![image](https://user-images.githubusercontent.com/35312807/227893692-d4616942-97d1-4c1d-bafb-5e77fd672bb3.png)

is it because I am black

**Comment by Xyncgas** [comment]
So turns out F# when an exception is thrown inside an asynchronous computation expression (CE), it is not caught by a try statement that surrounds the CE. This is because asynchronous computations are executed on a separate thread or thread pool, and the exception is thrown on that thread, which is outside the scope of the try block.

To catch exceptions in an asynchronous computation expression, you can use the Async.Catch function. This function takes an asynchronous computation expression as an argument and returns a new asynchronous computation expression that catches any exceptions thrown by the original computation.

**Comment by Xyncgas** [comment]
It's still not working if I use Async.Catch

**Comment by En3Tho** [comment]
Can you share a full code sample? Have you tried using native F# task CE's?
